src/neo/Network/P2P/Payloads/NotaryRequest.cs

```diff
@@ -0,0 +1,143 @@
+        {
+            DeserializeUnsigned(reader);
+            witness = reader.ReadSerializable<Witness>();
+        }
+
+        public void Serialize(BinaryWriter writer)
+        {
+            SerializeUnsigned(writer);
+            writer.Write(witness);
+        }
+
+        public void SerializeUnsigned(BinaryWriter writer)
+        {
+            writer.Write(mainTransaction);
+            writer.Write(fallbackTransaction);
+        }
+
+        public UInt160[] GetScriptHashesForVerifying(DataCache snapshot)
+        {
+            return new UInt160[] { fallbackTransaction.Signers[1].Account };
+        }
+
+        public bool VerifyStateIndependent(ProtocolSettings settings)
+        {
+            var nKeysMain = mainTransaction.GetAttributes<NotaryAssisted>();
+            if (!nKeysMain.Any()) return false;
+            if (nKeysMain.ToArray()[0].NKeys == 0) return false;
+            if (fallbackTransaction.Signers.Length != 2) return false;
+            if (fallbackTransaction.Witnesses[0].InvocationScript.Length != 66
+                || fallbackTransaction.Witnesses[0].VerificationScript.Length != 0
+                || fallbackTransaction.Witnesses[0].InvocationScript[0] != (byte)OpCode.PUSHDATA1 || fallbackTransaction.Witnesses[0].InvocationScript[1] != 64)
+                return false;
+            if (fallbackTransaction.Sender != NativeContract.Notary.Hash) return false;
+            if (fallbackTransaction.Signers[1].Account != witness.ScriptHash) return false;
+            if (fallbackTransaction.GetAttribute<NotValidBefore>() is null) return false;
+            var conflicts = fallbackTransaction.GetAttributes<ConflictAttribute>();
+            if (conflicts.Count() != 1) return false;
+            if (conflicts.ToArray()[0].Hash != mainTransaction.Hash) return false;
+            var nKeysFallback = fallbackTransaction.GetAttributes<NotaryAssisted>();
+            if (!nKeysFallback.Any()) return false;
+            if (nKeysFallback.ToArray()[0].NKeys != 0) return false;
+            if (mainTransaction.ValidUntilBlock != fallbackTransaction.ValidUntilBlock) return false;
+            if (!fallbackTransaction.VerifyPartialStateIndependent(settings)) return false;
+            if (!mainTransaction.VerifyPartialStateIndependent(settings, false)) return false;
+            return this.VerifyWitnesses(settings, null, MaxVerificationGas);
+        }
+
+        public bool VerifyStateDependent(ProtocolSettings settings, DataCache snapshot)
+        {
+            var balance = NativeContract.Notary.BalanceOf(snapshot, fallbackTransaction.Signers[1].Account);
+            if (fallbackTransaction.NetworkFee + fallbackTransaction.SystemFee > balance)
+                return false;
+            if (!fallbackTransaction.VerifyPartialStateDependent(settings, snapshot)) return false;
+            if (!mainTransaction.VerifyPartialStateDependent(settings, snapshot)) return false;
```

Comment:
Main transaction should not be verified. We care only about fallback. Moreover, it won't pass verification because it doesn't have all witnesses attached.